view/frontend/templates/button_product_page.phtml

```diff
@@ -0,0 +1,278 @@
+                    // and calls the final Bolt defined callback
+                    var onSuccess = function(data){
+                        if (data.status !== 'success') {
+                            if (data.message) {
+                                showBoltErrorMessage();
+                                // pretend order creation was success...
+                                // we need to call this; otherwise bolt modal show infinte spinner.
+                                callback();
+                            }
+
+                            return false;
+                        }
+                        try {
+                            <?php echo $block->getjavascriptSuccess(); ?>
+                            callbacks.success_url = data.success_url;
+
+                            trackCallbacks.onSuccess();
+                        } finally {
+                            callback();
+                        }
+                    };
+                    // ajax call to the update order transaction data endpoint.
+                    // passes the bolt transaction reference
+                    save_request = ajaxPost(settings.save_order_url, parameters, onSuccess);
+                },
+
+                check: function () {
+                    /**
+                     * On Bolt button click check if guest checkout is allowed.
+                     * Display login popup to guest customers if it is not. The
+                     * Magento customerData and authenticationPopup objects are
+                     * used.
+                     */
+
+                    //check if we have inventory on stock
+                    if ( (max_qty>=0) && $qty && (Number($qty.val()) > max_qty)) {
+                        window.alert("Please correct the quantity. The most you may purchase is " + max_qty + ".");
+                        return false;
+                    }
+
+                    // check if login is required
+                    if ( !customer().firstname && !isGuestCheckoutAllowed) {
+                        // if authentication is required for checkout set a cookie
+                        // for auto opening Bolt checkout after login
+                        authenticationPopup.showModal();
+                        return false;
+                    }
+                    return true;
+                },
+
+                onEmailEnter: function(email) {
+                    trackCallbacks.onEmailEnter(email);
+                    if (callbacks.email !== email) {
+                        ajaxPost(settings.save_email_url, 'email='+encodeURIComponent(email));
+                        callbacks.email = email;
+                    }
+                }
+            };
+
+            function setupProductPage() {
+                var quantity = 1;
+                if ($qty) {
+                    quantity = Number($qty.val());
+                }
+                var cart = {
+                    currency: "USD",
```

Comment:
There's a recent plugin update to support multi currency. The currency is actually read from quote in getCartData. We don't have the quote yet. I assume the currency must be available on the product page. Can you try to load it instead od hard coding "USD"?

Reply:
@danac-gs Let me do it in separate PR, I've created asana https://app.asana.com/0/0/1155768935724793/f